packages/nodejs/src/transmitter.ts

```diff
@@ -0,0 +1,96 @@
+import fs from "fs"
+import path from "path"
+import https from "https"
+import http from "http"
+
+import { Configuration } from "./config"
+import { URL, URLSearchParams } from "url"
+import { HashMap } from "@appsignal/types"
+
+export class Transmitter {
+  #config: Configuration
+  #url: string
+  #data: string
+
+  constructor(url: string, data = "", active = true) {
+    this.#config = new Configuration({ active })
+    this.#url = url
+    this.#data = data
+  }
+
+  public async transmit() {
```

Comment:
If we return a promise, we don't need to make it async. We also don't call `await` anywhere in the function so I think we can remove it. That also means you can remove `await` anywhere we call `transmitter.transmit()`

This is a similar function that doesn't need async: https://github.com/appsignal/appsignal-nodejs/blob/402a364305777bda77fc71b9114d0429595b2cd5/packages/nodejs-ext/scripts/extension.js#L44-L45

Reply:
It has to be async because of the `validatePushApiKey()` function. It returns promises based on the result of `transmit()` and if we don't call it with `await`, it always returns `undefined`.

I removed the `await` call from `sendReport()`

Reply:
I'm still confused about the whole async-await in combination with promises, but I think what I said before is wrong, in this context. If we don't await on `transmitter.transmit()` in `sendReport` it continues the program execution, but because it's the last thing it does, it waits on the promise before exiting. I think calling await on everywhere that we call `transmitter.transmit()` was correct before. 

Every async function returns a promise, even if we don't explicitly wrap the return value in one apparently.
I read that one would usually not need both async-await and promises with `then`-calls but I don't see how just yet, we also need to "catch" failure states.